src/Sarif/Schemata/sarif-schema.json

```diff
@@ -2388,6 +2399,7 @@
           "type": "array",
           "minItems": 0,
           "uniqueItems": false,
+          "default": [],
```

Comment:
This is definitely wrong. The default is null. A non-empty array is an explicit indicator that a run succeeded, but there were no results. A null value indicates that the analysis was not able to be completed for some reason. @lgolding, did you notice this during your review? Harlene says the spec sets this default (empty array). That seems wrong to me.

Reply:
I did notice it but I got confused. In the PR's summary comment, I wrote:

- run.results should be empty array [] by default (@harleenkohli -- it's actually `externalPropertyFileReferences.results` whose default is `[]`, not `run.results`. But the code you wrote is correct. -- Larry)

So you see I knew that `run.results` shouldn't default to empty array. But when I said "the code you wrote is correct", it's because I thought this line _was_ in the `externalPropertyFileReferences` object. I was wrong; it's in the `run` object, and it needs to be moved to the `externalPropertyFileReferences` object.

Now as to what the spec says -- I'll check, and bug fix it if it says the default is `[]`.

---
In reply to: [275445302](https://github.com/Microsoft/sarif-sdk/pull/1426#discussion_r275445302) [](ancestors = 275445302)

Reply:
Closing the loop: the spec does not specify a default for `run.results`, nor does it for any other object-valued property.

---
In reply to: [275452387](https://github.com/Microsoft/sarif-sdk/pull/1426#discussion_r275452387) [](ancestors = 275452387,275445302)